undertow/src/main/java/org/wildfly/extension/undertow/filters/ModClusterBalancerDefinition.java

```diff
@@ -35,13 +46,133 @@
+                context.getResult().set(new ModelNode(ctx.getMaxAttempts()));
+            }
+        });
+        resourceRegistration.registerReadOnlyAttribute(STICKY_SESSION_COOKIE, new AbstractBalancerOperation() {
+
+            @Override
+            protected void handleNode(OperationContext context, ModClusterStatus.LoadBalancer ctx, ModelNode operation) throws OperationFailedException {
+                context.getResult().set(new ModelNode(ctx.getStickySessionCookie()));
+            }
+        });
+
+        resourceRegistration.registerReadOnlyAttribute(STICKY_SESSION_PATH, new AbstractBalancerOperation() {
+
+            @Override
+            protected void handleNode(OperationContext context, ModClusterStatus.LoadBalancer ctx, ModelNode operation) throws OperationFailedException {
+                context.getResult().set(new ModelNode(ctx.getStickySessionPath()));
+            }
+        });
+
+        resourceRegistration.registerReadOnlyAttribute(WAIT_WORKER, new AbstractBalancerOperation() {
+
+            @Override
+            protected void handleNode(OperationContext context, ModClusterStatus.LoadBalancer ctx, ModelNode operation) throws OperationFailedException {
+                context.getResult().set(new ModelNode(ctx.getWaitWorker()));
+            }
+        });
+
+        resourceRegistration.registerReadOnlyAttribute(STICKY_SESSION, new AbstractBalancerOperation() {
+
+            @Override
+            protected void handleNode(OperationContext context, ModClusterStatus.LoadBalancer ctx, ModelNode operation) throws OperationFailedException {
+                context.getResult().set(new ModelNode(ctx.isStickySession()));
+            }
+        });
+        resourceRegistration.registerReadOnlyAttribute(STICKY_SESSION_REMOVE, new AbstractBalancerOperation() {
+
+            @Override
+            protected void handleNode(OperationContext context, ModClusterStatus.LoadBalancer ctx, ModelNode operation) throws OperationFailedException {
+                context.getResult().set(new ModelNode(ctx.isStickySessionRemove()));
+            }
+        });
+        resourceRegistration.registerReadOnlyAttribute(STICKY_SESSION_FORCE, new AbstractBalancerOperation() {
+
+            @Override
+            protected void handleNode(OperationContext context, ModClusterStatus.LoadBalancer ctx, ModelNode operation) throws OperationFailedException {
+                context.getResult().set(new ModelNode(ctx.isStickySessionForce()));
+            }
+        });
+    }
+
+
+    private abstract class AbstractBalancerOperation implements OperationStepHandler {
+
+        @Override
+        public final void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+            PathAddress address = PathAddress.pathAddress(operation.get(ModelDescriptionConstants.ADDRESS));
+            int current = address.size() - 1;
+            String balancerName = address.getElement(current--).getValue();
```

Comment:
could be simplified with
PathAddress address = context.getCurrentAddress();
String balancerName = address.getParent().getValue();
...